Мордвинцева, HW2

mordvintseva opened the pull request.

Commit:
First homework. Initial commit

Commit:
Add .gitignore

Commit:
Code refactoring

Commit:
Change of characters loading

Character loading when reaching the end of UITableView
Delete initializer of Character struct

Commit:
Delete variable category from CharacterViewController

Commit:
Clear repository on new branch

Commit:
Add .gitignore

Commit:
Initial commit

Commit:
Fix main functions of the application

Database has been established
Add image display on CharacterViewController

M0rtyMerr commented on Mordvintseva/Rick-and-Morty/Rick-and-Morty.xcodeproj/project.pbxproj:
```diff
@@ -0,0 +1,544 @@
+			);
+			runOnlyForDeploymentPostprocessing = 0;
+			shellPath = /bin/sh;
+			shellScript = "#swiftlint autocorrect\n#if which swiftlint >/dev/null; then\n#swiftlint\n#else\n#echo \"warning: SwiftLint not installed, download from https://github.com/realm/SwiftLint\"\n#fi\n";
```
Это смахивает на закоменченные строчки. Помочь наладить или еще раз readme.md перечитаешь?)

mordvintseva replied:
Перечитаю(

M0rtyMerr replied:
Ты вставила строчки для другого способа установки, нужные тебе - чуть выше

M0rtyMerr commented on Mordvintseva/Rick-and-Morty/Rick-and-Morty/AppDelegate.swift:
```diff
@@ -0,0 +1,19 @@
+    var window: UIWindow?
+
+    // swiftlint:disable:next line_length
+    func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?) -> Bool {
```
Если ничего здесь не делаешь, можешь просто удалять

M0rtyMerr commented on Mordvintseva/Rick-and-Morty/Rick-and-Morty/Screen/CharacterView/CharacterViewController.swift:
```diff
@@ -0,0 +1,62 @@
+        self.title = character.name
+        self.infoTable.dataSource = self
+        self.infoTable.tableFooterView = UIView()
+        let minSide = min(image.frame.height, image.frame.height)
```
Смотрим что меньше, высота или высота. Не совсем понял

mordvintseva replied:
Ошиблась, надо выбирать меньшее из ширины/высоты

M0rtyMerr replied:
А зачем?

mordvintseva replied:
Попробовала посмотреть, как отображается экран на разных устройствах, получилось так, что где-то ширина меньше высоты, где-то наоборот... Наверное, есть другие способы скорректировать изображение

M0rtyMerr replied:
Ну так коненчо работает, но вообще скейл мод и констрейнты. Завтра буду подробнее рассказывать

M0rtyMerr commented on Mordvintseva/Rick-and-Morty/Rick-and-Morty/Screen/CharacterView/CharacterViewController.swift:
```diff
@@ -0,0 +1,62 @@
+        image.getImageByURL(url: URL(string: character.image)!, size: CGSize(width: minSide, height: minSide))
+    }
+
+    private enum CharacterInfo: Int, CaseIterable {
```
Все вложенные типы принято объявлять наверху, чтобы они не путались с функциями и переменными

M0rtyMerr commented on Mordvintseva/Rick-and-Morty/Rick-and-Morty/Screen/MainPage/Cell/CharacterCell.swift:
```diff
@@ -0,0 +1,38 @@
+    func set(name: String, imageURL: String) {
+        guard let imageURL = URL(string: imageURL) else { return }
+        nameLabel.text = name
+        avatar.getImageByURL(url: imageURL, size: CGSize(width: avatar.frame.width, height: avatar.frame.height))
```
Тут есть небольшая проблема. Ячейки переиспользуются и каждый раз метод set вызывается. И отправляется новый запрос на картинку, старый при этом НЕ отменяется. Могут быть баги с этим
Нужно отменять предыдущий запрос в func prepareForReuse с помощью avatar.kf.cancelDownloadTask()

M0rtyMerr commented on Mordvintseva/Rick-and-Morty/Rick-and-Morty/Screen/MainPage/Cell/CharacterCell.swift:
```diff
@@ -0,0 +1,38 @@
+    }
+
+    func resizeImage(image: UIImage, width: CGFloat, height: CGFloat) -> UIImage {
+        UIGraphicsBeginImageContextWithOptions(CGSize(width: width, height: height), true, 0.0)
```
А еще можно просто поменять у UIImageView скейл мод. Пробовала так?

mordvintseva replied:
Нет, но попробую

M0rtyMerr commented on Mordvintseva/Rick-and-Morty/Rick-and-Morty/Screen/MainPage/Cell/CharacterCell.swift:
```diff
@@ -0,0 +1,38 @@
+        return newImage!
+    }
+
+    override func awakeFromNib() {
```
Пустые методы убираем

M0rtyMerr commented on Mordvintseva/Rick-and-Morty/Rick-and-Morty/Screen/MainPage/MainPageController.swift:
```diff
@@ -0,0 +1,85 @@
+import Alamofire
+
+class MainPageController: UIViewController, UITableViewDelegate {
+    @IBOutlet weak var tableView: UITableView!
```
Swiftlint бы здесь поругал

MortyMerr commented:
Все хорошо. Даже констрейнты есть, footer поставила. Но не хватает чего-то вот знаешь такого... клевого. Что нибудь новое найти и показать что умеешь. LargeTitles там сделать или поиск. Или свою кнопку назад для UINavigationController. Таблицы ты умеешь делать, но останавливаться в изысканиях нельзя :)

MortyMerr commented:
Отдельный лайк за английские и членоразделные commit message

Commit:
Code correction

Delete resizeImage in CharacterCell
Delete empty methods
Add aspect ratio to image in CharacterViewController

MortyMerr commented:
Молодец, стало лучще

MortyMerr commented:
@mordvintseva ну а где калькулятор или scrollView, алле
